packages/frontend/src/pages/Executions/index.tsx

```diff
@@ -1,73 +1,167 @@
 import type { IExecution } from '@plumber/types'
 
-import * as React from 'react'
-import { Link, useSearchParams } from 'react-router-dom'
+import { ChangeEvent, ReactElement, useEffect, useRef, useState } from 'react'
+import { BiSearch } from 'react-icons/bi'
+import { useSearchParams } from 'react-router-dom'
 import { useQuery } from '@apollo/client'
-import Box from '@mui/material/Box'
-import CircularProgress from '@mui/material/CircularProgress'
-import Divider from '@mui/material/Divider'
-import Grid from '@mui/material/Grid'
+import {
+  Box,
+  CircularProgress,
+  Divider,
+  Flex,
+  Icon,
+  Input,
+  InputGroup,
+  InputLeftElement,
+  InputRightElement,
+} from '@chakra-ui/react'
 import Pagination from '@mui/material/Pagination'
-import PaginationItem from '@mui/material/PaginationItem'
 import Container from 'components/Container'
 import ExecutionRow from 'components/ExecutionRow'
+import ExecutionStatusMenu, { StatusType } from 'components/ExecutionStatusMenu'
 import NoResultFound from 'components/NoResultFound'
 import PageTitle from 'components/PageTitle'
 import { GET_EXECUTIONS } from 'graphql/queries/get-executions'
 import useFormatMessage from 'hooks/useFormatMessage'
+import debounce from 'lodash/debounce'
 
 const EXECUTION_PER_PAGE = 10
 const EXECUTIONS_TITLE = 'Executions'
 
-const getLimitAndOffset = (page: number) => ({
+interface ExecutionParameters {
+  page: number
+  status: string
+  input: string
+}
+
+const getLimitAndOffset = (params: ExecutionParameters) => ({
   limit: EXECUTION_PER_PAGE,
-  offset: (page - 1) * EXECUTION_PER_PAGE,
+  offset: (params.page - 1) * EXECUTION_PER_PAGE,
+  ...(params.status !== StatusType.Waiting && { status: params.status }),
+  searchInput: params.input,
 })
 
-export default function Executions(): React.ReactElement {
+export default function Executions(): ReactElement {
   const formatMessage = useFormatMessage()
   const [searchParams, setSearchParams] = useSearchParams()
   const page = parseInt(searchParams.get('page') || '', 10) || 1
 
-  const { data, refetch, loading } = useQuery(GET_EXECUTIONS, {
-    variables: getLimitAndOffset(page),
+  const [searchInput, setSearchInput] = useState<string>(
+    searchParams.get('input') || '',
+  )
+  const [filterStatus, setFilterStatus] = useState<string>(
+    searchParams.get('status') || '',
+  )
```

Comment:
we dont need to maintain a separate state since it's actually 'maintained' in search params already.
i.e. u can just read from
const searchInput = searchParams.get('input') || ''
const filterStatus = searchParams.get('status') || ''

Reply:
wah nice thanks